### DISCUSSION TITLE
Document Set Default View

### DISCUSSION THREAD

**Discussion by paylord** [pnp/powershell#2181] [Q&A]
Is it possible to set the default view within a Document Set using PnP?

If not, can this be added as an enhancement request?

**Comment by SabbeRubbish** [comment]
Would also see the benefit to this.
e.g. via Set-PnPDocumentSetField -DefaultView

**Comment by markgort86** [comment]
For my opinion this is possible by using the following cmdlet: `Set-PnPView`. See also: https://pnp.github.io/powershell/cmdlets/Set-PnPView.html. There is a property for DefaultView. If you change this to true, the view will be the default.

**Reply by SabbeRubbish** [reply]
That would set the default view for the entire library/list - I think.
Not sure it would set the document set default view.

**Reply by markgort86** [reply]
Yes, it will change the default view for the whole list. But how do i reach this scenario by just using the user interface within SharePoint Online on a modern site where a document set is available? There are no view options available in the "DOCUMENT SET SETTINGS" page.

**Reply by SabbeRubbish** [reply]
In the document set settings page, at the bottom, you find the Document Set Welcome Page view:
![image](https://github.com/pnp/powershell/assets/5593135/ffe116e8-a79d-459c-a9c2-b706c53db7c6)

Yes, the welcome page isn't available in Modern SharePoint, but that view is still used when you navigate to a document set in Modern SharePoint.
And this makes a lot of sense.
Say you have 6 fields in your document set like owner, start/end date, status, ... It's logical to show this in the default view for the **list**. However, when you click on 1 document set, all fields for all documents below it are typically synchronized with the document set, having no need to show all of those columns, hence showing a different view makes a lot of sense there.

**Comment by paylord** [comment]
![image](https://github.com/pnp/powershell/assets/8131698/8ac6516d-3831-483d-b647-f0ad3ec0968b)

Like this?

**Reply by SabbeRubbish** [reply]
Yes exactly, like my other post suggested :)

**Reply by SabbeRubbish** [reply]
Any update on this?

**Reply by crawoo** [reply]
An update on this would be great! Trying to do the very same thing now.

**Reply by SabbeRubbish** [reply]
Unfortunately, it seems like that element is the only one on that configuration page that does not have a REST counterpart.
On premises you can update the SchemaXml of the content type, but that doesn't work in SharePoint Online.

I've thought of 2 workarounds:
- Using web automation to click through the pages (see [my blogpost](https://smindreau.wordpress.com/2024/11/20/automating-document-set-welcome-page-view-updates/) about this)
- A though had occurred to me that we might be able to use column formatting to create the link ourselves on document sets. Probably best to do this with a SPFx customization as I think the approach with JSON may be too cumbersome (not having all necessary fields loaded in the view as just the first hurdle I see coming)

**Reply by MrRCG5** [reply]
Is there any updates on changing the **Doc Set > Welcome Page View** via PowerShell or Power Automate?

**Reply by SabbeRubbish** [reply]
Would really appreciate a decent workaround or API update here, but I guess we should be ringing Microsoft's door for this?
Any clue on where we should post this request?

**Reply by SethiaGit1988** [reply]
i am also looking for updating document set welcome view, can this be picked up earliest, as at the moment, you can only do it though UI, and we have thousand of sites where we need to automate it

**Comment by SethiaGit1988** [comment]
any update on this issue please, there is anyway we could update the document set welcome page view

**Reply by SabbeRubbish** [reply]
Hi Sethia, your best bet right now is web automation - see my [blogpost](https://smindreau.wordpress.com/2024/11/20/automating-document-set-welcome-page-view-updates/) for details. 
It's far from great, but better than going through thousands of sites by hand :-)

But I will support your shoutout to the SharePoint team to include this in their APIs, even though they're probably never going to read this :D
